### PR TITLE
chore(doc): generate all docs

### PR DESCRIPTION
**Pull request by sguiheux**

### CODE REVIEW

File: sdk/doc/generate.go [hunk 1]
```diff
@@ -99,3 +108,222 @@ func genMarkdownTreeCustom(cmd *cobra.Command, rootdir string, filePrepender, li
 	cmd.DisableAutoGenTag = true
 	return doc.GenMarkdownCustom(cmd, f, linkHandler)
 }
+
```
Comment (fsamin) [inline sdk/doc/generate.go]: could you add an example of how we have to write doc now ?

File: sdk/doc/generate.go [hunk 2]
```diff
@@ -2,23 +2,27 @@ package doc
...
+	"go/parser"
+	"go/token"
 	"io"
 	"os"
 	"path"
 	"path/filepath"
 	"strings"
 
+	"github.com/ovh/cds/sdk"
 	"github.com/spf13/cobra"
```
Comment (bnjjj) [inline sdk/doc/generate.go]: blank line

File: docs/content/contribute/_index.md [hunk 3]
```diff
@@ -37,9 +37,8 @@ Documentation https://ovh.github.io/cds is generated with Hugo. Source are under
 ```bash
 cd ${CDS_SOURCES}/docs/content/cli
 rm -rf cdsctl engine worker;
-cdsctl doc  # generate cdsctl directory documentation in the current directory so you must be inside ${CDS_SOURCES}/docs/content/cli
-engine doc  # generate engine directory documentation in the current directory so you must be inside ${CDS_SOURCES}/docs/content/cli
-worker doc  # generate worker directory documentation in the current directory so you must be inside ${CDS_SOURCES}/docs/content/cli
+cd ${CDS_SOURCES}
+GEN_PATH=${CDS_SOURCES}/docs/content/cli make doc
```
Comment (bnjjj) [inline docs/content/contribute/_index.md]: need go install worker, engine and cdsctl I think we must indicate this part in this example script

File: sdk/doc/generate.go (hunk 1 above)
Comment (yesnault) [reply sdk/doc/generate.go]: done, cf. https://github.com/ovh/cds/pull/2012/commits/778af69839075f610e779b8f4595ac5ed964ea20

**Comment by yesnault** [comment]
example: 
capture d ecran 2018-01-22 a 23 20 31

**Comment by yesnault** [comment]
@fsamin 
with:

```
// getActionsHandler Retrieve all public actions
// @title List all public actions
// @description the description
// @params AA=valA
// @params BB=valB
// @requestBody {"mykey": "myval"}
// @responseBody {"mykey": "myval"}
```

capture d ecran 2018-01-23 a 22 39 12

**Pull request merged**